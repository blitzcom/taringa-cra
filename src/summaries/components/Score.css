.Score-display {
  border-right: 1px solid #f0f0f0;
  padding-right: 8px;
  text-align: center;
  margin-right: 16px;
}

.Score-vote {
  background: none;
  border: none;
  color: #b6b6b6;
  cursor: pointer;
}

.Score-disabled {
  color: #f0f0f0 !important;
  cursor: default !important;
}

.Score-disabled:hover {
  color: #f0f0f0 !important;
  cursor: default !important;
}

.Score-vote:focus,
.Score-vote:active,
.Score-vote:hover {
  outline: none;
}

.Score-vote:active,
.Score-vote:hover {
  color: #262626;
}

.Score-votes {
  font-size: .8rem;
  font-weight: bold;
}

.Score-up,
div.Score-up {
  color: #00e000;
}

.Score-down,
div.Score-down {
  color: #ff0000;
}
