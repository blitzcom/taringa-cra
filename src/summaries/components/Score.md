Basic Score

```js
<Score downvotes={2500} upvotes={5500} />
```

Voted up

```js
<Score downvotes={2500} upvotes={5500} voted={1} />
```

Voted down

```js
<Score downvotes={2500} upvotes={5500} voted={-1} />
```


Voting

```js
<Score downvotes={2500} upvotes={5500} isVoting />
```
