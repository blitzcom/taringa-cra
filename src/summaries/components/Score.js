import React from 'react'
import classNames from 'classnames'
import _ from 'lodash'

import './Score.css'
import { humanizeNum } from '../../Utils'

const Score = props => {
  const classVoteUpButton = classNames('Score-vote', {
    'Score-disabled': props.isVoting,
    'Score-up': props.voted > 0,
  })

  const classVoteDownButton = classNames('Score-vote', {
    'Score-disabled': props.isVoting,
    'Score-down': props.voted < 0,
  })

  const classVotes = classNames('Score-votes', {
    'Score-disabled': props.isVoting,
    'Score-down': props.voted < 0,
    'Score-up': props.voted > 0,
  })

  return (
    <div className="Score-display">
      <button
        className={classVoteUpButton}
        disabled={props.isVoting}
        onClick={() => {
          if (!props.isVoting) {
            props.onVoteUp()
          }
        }}
      >
        <i className="fa fa-arrow-up"/>
      </button>

      <div className={classVotes}>
        {(_.isInteger(props.upvotes) && _.isInteger(props.downvotes))
          ? humanizeNum(props.upvotes - props.downvotes)
          : 0}
      </div>

      <button
        className={classVoteDownButton}
        disabled={props.isVoting}
        onClick={() => {
          if (!props.isVoting) {
            props.onVoteDown()
          }
        }}
      >
        <i className="fa fa-arrow-down"/>
      </button>
    </div>
  )
}

Score.defaultProps = {
  downvotes: 0,
  isVoting: false,
  onVoteDown: () => {},
  onVoteUp: () => {},
  upvotes: 0,
  voted: 0,
}

export default Score
