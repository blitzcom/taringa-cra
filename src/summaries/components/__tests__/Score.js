import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import Score from '../Score'

describe('Score', () => {
  it('renders default', () => {
    const tree = renderer
      .create(<Score downvotes={2500} upvotes={3500} />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('voted up', () => {
    const tree = renderer
      .create(<Score downvotes={2500} upvotes={3500} voted={1} />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('voted down', () => {
    const tree = renderer
      .create(<Score downvotes={2500} upvotes={3500} voted={-1} />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('voting', () => {
    const tree = renderer
      .create(<Score downvotes={2500} upvotes={3500} isVoting />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('handles onVoteUp', () => {
    const mock = jest.fn()
    const score = shallow(<Score onVoteUp={mock}/>)

    score.find('button').first().simulate('click')

    expect(mock).toBeCalled()
  })

  it('handles onVoteDown', () => {
    const mock = jest.fn()
    const score = shallow(<Score onVoteDown={mock}/>)

    score.find('button').last().simulate('click')

    expect(mock).toBeCalled()
  })

  it('cannot vote if is voting', () => {
    const up = jest.fn()
    const down = jest.fn()
    const score = shallow(<Score isVoting onVoteDown={down} onVoteUp={up}/>)

    score.find('button').last().simulate('click')
    score.find('button').first().simulate('click')

    expect(up).not.toBeCalled()
    expect(down).not.toBeCalled()
  })
})
