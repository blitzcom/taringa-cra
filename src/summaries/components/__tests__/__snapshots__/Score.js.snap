// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Score renders default 1`] = `
<div
  className="Score-display"
>
  <button
    className="Score-vote"
    disabled={false}
    onClick={[Function]}
  >
    <i
      className="fa fa-arrow-up"
    />
  </button>
  <div
    className="Score-votes"
  >
    1k
  </div>
  <button
    className="Score-vote"
    disabled={false}
    onClick={[Function]}
  >
    <i
      className="fa fa-arrow-down"
    />
  </button>
</div>
`;

exports[`Score voted down 1`] = `
<div
  className="Score-display"
>
  <button
    className="Score-vote"
    disabled={false}
    onClick={[Function]}
  >
    <i
      className="fa fa-arrow-up"
    />
  </button>
  <div
    className="Score-votes Score-down"
  >
    1k
  </div>
  <button
    className="Score-vote Score-down"
    disabled={false}
    onClick={[Function]}
  >
    <i
      className="fa fa-arrow-down"
    />
  </button>
</div>
`;

exports[`Score voted up 1`] = `
<div
  className="Score-display"
>
  <button
    className="Score-vote Score-up"
    disabled={false}
    onClick={[Function]}
  >
    <i
      className="fa fa-arrow-up"
    />
  </button>
  <div
    className="Score-votes Score-up"
  >
    1k
  </div>
  <button
    className="Score-vote"
    disabled={false}
    onClick={[Function]}
  >
    <i
      className="fa fa-arrow-down"
    />
  </button>
</div>
`;

exports[`Score voting 1`] = `
<div
  className="Score-display"
>
  <button
    className="Score-vote Score-disabled"
    disabled={true}
    onClick={[Function]}
  >
    <i
      className="fa fa-arrow-up"
    />
  </button>
  <div
    className="Score-votes Score-disabled"
  >
    1k
  </div>
  <button
    className="Score-vote Score-disabled"
    disabled={true}
    onClick={[Function]}
  >
    <i
      className="fa fa-arrow-down"
    />
  </button>
</div>
`;
